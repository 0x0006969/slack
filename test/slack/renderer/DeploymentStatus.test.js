const { DeploymentStatus } = require('../../../lib/slack/renderer/deployment-status');
const deploymentStatusPendingFixture = require('../../fixtures/webhooks/deployment/status_pending.json');
const deploymentStatusSuccessFixture = require('../../fixtures/webhooks/deployment/status_success.json');

describe('Deployment status rendering', () => {
  test('works for pending status', async () => {
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus: deploymentStatusPendingFixture.deployment_status,
      deployment: deploymentStatusPendingFixture.deployment,
      repository: deploymentStatusPendingFixture.repository,
    });
    expect(deploymentStatusMessage.getRenderedMessage()).toMatchSnapshot();
  });

  test('works for success status', async () => {
    const deploymentStatusMessage = new DeploymentStatus({
      deploymentStatus: deploymentStatusSuccessFixture.deployment_status,
      deployment: deploymentStatusSuccessFixture.deployment,
      repository: deploymentStatusSuccessFixture.repository,
    });
    expect(deploymentStatusMessage.getRenderedMessage()).toMatchSnapshot();
  });
});
