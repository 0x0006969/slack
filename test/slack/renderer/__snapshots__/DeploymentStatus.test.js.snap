// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deployment status rendering works for pending status 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#dbab09",
  "fallback": "Deploying 3dce5c8 to github-slack-app",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Deploying <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app|github-slack-app>",
}
`;

exports[`Deployment status rendering works for success status 1`] = `
Object {
  "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
  "author_link": "https://github.com/wilhelmklopp",
  "author_name": "wilhelmklopp",
  "color": "#28a745",
  "fallback": "Successfully deployed 3dce5c8 to github-slack-app",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Successfully deployed <https://github.com/github-slack/app/commit/3dce5c8488afc87a1ffbb840fc33b4a8331a1034|\`3dce5c8\`> to <https://dashboard.heroku.com/apps/github-slack-app/activity/builds/056df7bf-83c4-4ac1-911b-c8642413eef0|github-slack-app>",
}
`;
